landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+
+     Parameters
+     ----------
+     soil_density : float list
+         Density of soil in watershed (kg/m^3)
+     water_density : float
+         density of water in waterhed (kg/m^3)
+     C : float
+         Numerator of the debris flow threshold equation; Empirically
+         derived constant (See Tang et al. 2019)
+     N : float
+         Exponent for slope in the denominator of the debris flow
+         threshold equation; Empirically derived constant (See Tang et
+         al. 2019)
+
+     Examples
+     --------
+     >>> from landlab.components import DimensionlessDischarge
+     >>> from landlab import RasterModelGrid
+     >>> import random
+     >>> watershed_grid = RasterModelGrid((3, 3))
+     >>> flux = watershed_grid.add_ones('node', 'flux')
+     >>> d50 = watershed_grid.add_ones('node', 'd50')
+     >>> watershed_grid.at_node ['dem_values'] = \
+     ... np.array([[1.1, 2, 3, 4, 2, 3, 4, 5, 3]])
+     >>> dd = DimensionlessDischarge(watershed_grid)
+     >>> dd.run_one_step()
+     >>> print(watershed_grid.at_node['dimensionless_discharge'])
+     [ 0.55372743  0.55372743  0.55372743  0.55372743  0.55372743
+             0.55372743  0.55372743  0.55372743  0.55372743]
+
+
+
+     References
+     ----------
+     Tang, H., McGuire, L. A., Rengers, F. K., Kean, J. W., Staley,
+     D. M., & Smith, J. B. (2019). Developing and Testing Physically
+     Based Triggering Thresholds for Runoff-Generated Debris Flows.
+     Geophysical Research Letters, 46(15), 8830–8839.
+     https://doi.org/10.1029/2019GL083623
+
+    """
+
+    _name = "DimensionlessDischargeModel"
+
+    _unit_agnostic = False
+
+    _info = {
+        "dimensionless_discharge": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "flux in each stream segment",
+        },
+        "d50": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "soil grain size average in stream segment",
+        },
+        "dem_values": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "elevation stream segment",
+        },
+    }
```

Comment:
For new fields, please update the following two files:
docs/source/user_guide/field_definitions.rst
docs/source/user_guide/field_io.rst

The python script at the below link may be helpful. 
docs/build_standard_names_list.py

Reply:
I ran that script (docs/build_standard_names_list.py), but it didn't seem to do anything. What is it meant to do?

Reply:
Sorry, should have been clearer. That script can be used to generate the majority of the two files that need updating. 

I've updated the script so that it runs automatically when the docs are built and creates the entirety of the two files. (flat @mcflugen). Will push that change to this branch shortly.